mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -57,6 +57,21 @@ class VectorsSuite extends SparkFunSuite with Logging {
     assert(vec.values === values)
   }
 
+  test("sparse vector construction with mismatched indices/values array") {
+    intercept[IllegalArgumentException] {
+      Vectors.sparse(4, Array(1,2,3), Array(3.0, 5.0, 7.0, 9.0))
```

Comment:
Oh duh, fix coming. Every time I think I can't possibly need to run scalastyle as well as the test ...